### PR TITLE
feat: replace Get-Item powershell cmd with golang api on Windows HostProcess mode

### LINKED ISSUE #2436: reduce cpu, memory usage in CSI drivers on Windows
**Is your feature request related to a problem?/Why is this needed**

powershell command would consume more memory and cpu on Windows, it's better replace with Golang API in Windows host process mode
1. (Get-Item -Path $Env:mount).Target
    replace with `os.Readlink` does not work: https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/2265
2. there are other powershell commands in following code, it's better replacing them all with Golang APIs (search `RunPowershellCmd`)
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/pkg/os/disk/disk.go
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/pkg/os/volume/volume.go

**Describe the solution you'd like in detail**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by umagnus**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

feat: replace Get-Item powershell cmd with golang api on Windows HostProcess mode

**Which issue(s) this PR fixes**:

Fixes #2436 

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
feat: replace Get-Item powershell cmd with golang api on Windows HostProcess mode
```

### CODE REVIEW

File: pkg/os/volume/volume.go [hunk 1]
```diff
@@ -209,13 +210,12 @@ func getTarget(mount string, depth int) (string, error) {
 	if depth == 0 {
 		return "", fmt.Errorf("maximum depth reached on mount %s", mount)
 	}
-	cmd := "(Get-Item -Path $Env:mount).Target"
-	out, err := azureutils.RunPowershellCmd(cmd, fmt.Sprintf("mount=%s", mount))
-	if err != nil || len(out) == 0 {
-		return "", fmt.Errorf("error getting volume from mount. cmd: %s, output: %s, error: %v", cmd, string(out), err)
+	target, err := os.Readlink(mount)
```
Comment (andyzhangx) [inline pkg/os/volume/volume.go:213]: does `ReadLink` work? Have you tried manually on windows node?

(Get-Item -Path $Env:mount).Target
replace with os.Readlink does not work: https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/2265

Comment (umagnus) [reply pkg/os/volume/volume.go:213]: Yes, but it finally get \\?\ before Volume, so I also add this in condition
```
I1105 08:46:56.672088    9068 volume.go:218] mount c:\var\lib\kubelet\pods\b301374b-19e1-488b-9673-97ba23fc6530\volumes\kubernetes.io~csi\pvc-845f7fff-9798-4d28-afc3-361fcd7d58c9\mount is a symlink to c:\var\lib\kubelet\plugins\kubernetes.io\csi\disk.csi.azure.com\88391f4e3163e8a4a755a0a11f97af35d33fafb95860379f08761baa542ebf70\globalmount
I1105 08:46:56.672088    9068 volume.go:218] mount c:\var\lib\kubelet\plugins\kubernetes.io\csi\disk.csi.azure.com\88391f4e3163e8a4a755a0a11f97af35d33fafb95860379f08761baa542ebf70\globalmount is a symlink to \\?\Volume{23d86469-2073-4505-8db4-13d756102af8}\

```

Comment (umagnus) [reply pkg/os/volume/volume.go:213]: I also tried it on windows node for resize volume and it's ok

Comment (andyzhangx) [reply pkg/os/volume/volume.go:213]: what's the result of using `"(Get-Item -Path $Env:mount).Target"` ?

Comment (umagnus) [reply pkg/os/volume/volume.go:213]: only Volume{23d86469-2073-4505-8db4-13d756102af8} for final result

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**